frontend/routes/components/topMenu/components/visualSettingsDialog/visualSettingsDialog.component.tsx

```diff
@@ -0,0 +1,215 @@
+			</FormListItem>
+			<FormListItem>
+				XRay highlighting
+				<Field name="xray" render={ ({ field }) => (
+					<Switch checked={field.value} {...field} value="true" color="secondary" />)}/>
+			</FormListItem>
+		</List>);
+};
+
+const AdvancedSettings = (props) => {
+	return (
+		<List>
+			<FormListItem>
+				Show Statistics
+				<Field name="statistics" render={ ({ field }) => (
+					<Switch checked={field.value} {...field} value="true" color="secondary" />
+				)}/>
+			</FormListItem>
+			<FormListItem>
+				Memory for Unity
+				<Field name="memory" render={ ({ field, form }) => {
+					return (
+					<ErrorTooltip  title={form.errors.memory || ''} placement="bottom-end">
+					<ShortInput
+						error={Boolean(form.errors.memory)}
+						{...field}
+						endAdornment={<InputAdornment position="end">MB</InputAdornment>}/>
+					</ErrorTooltip>
+					);
+				}} />
+			</FormListItem>
+			<FormListItem>
+				Minimun near plane
+				<Field name="nearPlane" render={ ({ field, form }) => {
+					return (
+					<ErrorTooltip title={form.errors.nearPlane || ''} placement="bottom-end">
+					<ShortInput
+						error={Boolean(form.errors.nearPlane)}
+						{...field}
+						/>
+					</ErrorTooltip>
+					);
+				}} />
+			</FormListItem>
+			<FormListItem>
+				Far plane algorithm
+				<Field name="farPlaneAlgorithm" render={ ({ field }) => (
+					<Select {...field}>
+						<MenuItem value="box">Bounding box</MenuItem>
+						<MenuItem value="sphere">Bounding sphere</MenuItem>
+					</Select>
+				)}/>
+			</FormListItem>
+		</List>);
+};
+
+const Buttons = (props) => {
+	return (
+		<VisualSettingsButtonsContainer>
+			<Field render={ ({ form }) => (
+			<NegativeActionButton
+				color="primary"
+				variant="raised"
+				disabled={isEqual(form.values, DEFAULT_SETTINGS)}
+				type="button"
+				onClick={() => form.setValues(DEFAULT_SETTINGS)}
+				>
+					Reset
+				</NegativeActionButton>
+			)} />
+			<NeutralActionButton
+				color="primary"
+				variant="raised"
+				disabled={false}
+				type="button"
+				onClick={props.onClickCancel}
+			>
+				Cancel
+			</NeutralActionButton>
+			<Field render={ ({ form }) => (
+					<Button
+						color="secondary"
+						variant="raised"
+						type="submit"
+						disabled={!form.isValid || form.isValidating}
+						>
+						Save
+					</Button>
+			)} />
+
+		</VisualSettingsButtonsContainer>);
+};
+
+interface IProps {
+	handleResolve: () => void;
+	handleClose: () => void;
+	updateSettings: (settings: any) => void;
+	visualSettings: any;
+}
+
+interface IState {
+	selectedTab: number;
+	visualSettings: any;
+	flag: boolean;
+}
+
+export class VisualSettingsDialog extends React.PureComponent<IProps, IState> {
+	public state = {
+		selectedTab: 0,
+		visualSettings: null,
+		flag: false
+	};
+
+	public handleTabChange = (event, selectedTab) => {
+		this.setState({ selectedTab });
+	}
+
+	public onSubmit = (values, { resetForm }) => {
+		values.nearPlane = Number(values.nearPlane);
```

Comment:
I think you shouldn't mutate data here. Could you move this to `updateSettings` action?

Reply:
Sure, but why? Its just a conversion to pass the data to updateSettings with the correct datatypes, also its kind of the react's domain problem; which is that the forms output only strings.

Reply:
Doing this here, you potentially mutate the values object, which can cause some side effects.
Moreover, as you said, Formik returns always values as string. I think it's better to keep this behaviour unified across the app and change data type, when it's needed.

It's loose idea, so if you think it's not necessary just omit this comment.